services/base-static.js

```diff
@@ -4,53 +4,49 @@ const makeBadge = require('../gh-badges/lib/make-badge')
 const { makeSend } = require('../lib/result-sender')
 const analytics = require('../lib/analytics')
 const BaseService = require('./base')
-
-const serverStartTime = new Date(new Date().toGMTString())
+const {
+  serverHasBeenUpSinceResourceCached,
+  setCacheHeadersForStaticResource,
+} = require('./cache-headers')
 
 module.exports = class BaseStaticService extends BaseService {
-  // Note: Since this is a static service, it is not `async`.
-  handle(namedParams, queryParams) {
-    throw new Error(`Handler not implemented for ${this.constructor.name}`)
-  }
-
   static register({ camp }, serviceConfig) {
-    camp.route(this._regex, (queryParams, match, end, ask) => {
+    const {
+      profiling: { makeBadge: shouldProfileMakeBadge },
+    } = serviceConfig
```

Comment:
This is a teensy optimization. It avoids dotting into `serviceConfig.profiling.makeBadge` – twice! – every time this is called.

Reply:
What are the advantages over simply doing `const shouldProfileMakeBadge  = serviceConfig.profiling.makeBadge`?

Reply:
The difference is purely stylistic.